src/TestProblem/Hydro/CCSN/Init_TestProb_Hydro_CCSN.cpp

```diff
@@ -506,138 +527,35 @@ void Load_IC_Prof_CCSN()
-   const double BoxCenter[3]            = { amr->BoxCenter[0], amr->BoxCenter[1], amr->BoxCenter[2] };
-
-// allocate memory for per-thread arrays
-#  ifdef OPENMP
-   const int NT = OMP_NTHREAD;   // number of OpenMP threads
-#  else
-   const int NT = 1;
-#  endif
-
-   double DataCoord[4] = { -__DBL_MAX__ }, **OMP_DataCoord=NULL;
-   Aux_AllocateArray2D( OMP_DataCoord, NT, 4 );
-
-
-#  pragma omp parallel
-   {
-#     ifdef OPENMP
-      const int TID = omp_get_thread_num();
-#     else
-      const int TID = 0;
-#     endif
-
-//    initialize arrays
-      OMP_DataCoord[TID][0] = -__DBL_MAX__;
-      for (int b=1; b<4; b++)   OMP_DataCoord[TID][b] = 0.0;
-
-      for (int lv=0; lv<NLEVEL; lv++)
-      {
-         const double dh = amr->dh[lv];
-
-#        pragma omp for schedule( runtime )
-         for (int PID=0; PID<amr->NPatchComma[lv][1]; PID++)
-         {
-            if ( amr->patch[0][lv][PID]->son != -1 )  continue;
-
-            for (int k=0; k<PS1; k++)  {  const double z = amr->patch[0][lv][PID]->EdgeL[2] + (k+0.5)*dh;
-            for (int j=0; j<PS1; j++)  {  const double y = amr->patch[0][lv][PID]->EdgeL[1] + (j+0.5)*dh;
-            for (int i=0; i<PS1; i++)  {  const double x = amr->patch[0][lv][PID]->EdgeL[0] + (i+0.5)*dh;
-
-               const double dens = amr->patch[ amr->FluSg[lv] ][lv][PID]->fluid[DENS][k][j][i];
-
-               if ( dens > OMP_DataCoord[TID][0] )
-               {
-                  OMP_DataCoord[TID][0] = dens;
-                  OMP_DataCoord[TID][1] = x;
-                  OMP_DataCoord[TID][2] = y;
-                  OMP_DataCoord[TID][3] = z;
-               }
-
-            }}} // i,j,k
-         } // for (int PID=0; PID<amr->NPatchComma[lv][1]; PID++)
-      } // for (int lv=0; lv<NLEVEL; lv++)
-   } // OpenMP parallel region
-
-
-// find the maximum over all OpenMP threads
-   for (int TID=0; TID<NT; TID++)
-   {
-      if ( OMP_DataCoord[TID][0] > DataCoord[0] )
-         for (int b=0; b<4; b++)   DataCoord[b] = OMP_DataCoord[TID][b];
-   }
-
-// free per-thread arrays
-   Aux_DeallocateArray2D( OMP_DataCoord );
+// 1. record the maximum density
+   Record_CCSN_CentralDens();
 
 
-// collect data from all ranks
-#  ifndef SERIAL
+#ifdef GRAVITY
+// 2. record the GW signal
+   if ( CCSN_GW_OUTPUT )
    {
-      double DataCoord_All[4 * MPI_NRank];
+      static double DumpTime_Last = -__DBL_MAX__;
 
-      MPI_Allgather( DataCoord, 4, MPI_DOUBLE, DataCoord_All, 4, MPI_DOUBLE, MPI_COMM_WORLD );
+             double PhysicalTime = Time[0];
+             double DumpTime = int( PhysicalTime / CCSN_GW_DT ) * CCSN_GW_DT;
 
-      for (int i=0; i<MPI_NRank; i++)
+//    output data when (a) the elapsed time > GW_OUTPUT_DT
+//                     (b) the physical time is almost a multiple of GW_OUTPUT_DT
+      if ( fabs( PhysicalTime - DumpTime_Last ) > CCSN_GW_DT  ||
+           fabs( PhysicalTime - DumpTime      ) < 1.0e-2 * CCSN_GW_DT )
```

Comment:
I forgot why I did that. Since `CCSN_GW_DT` will not change the evolution time-step, probably the condition aims to  dump the data if `time-step < CCSN_GW_DT < 2 * time-step`. Now I use a simpler condition that dumping the data when evolution time-step > `CCSN_GW_DT`.